### PR TITLE
Update diagrams and Architecture section

### LINKED ISSUE #8: Nearly every diagram requires updating
Most diagrams are inaccurate.  Some contain mentions of things such as etcd, fleet, and deisctl.  Some express relationships between components that no longer exist.  For instance, the logger does not depend on the store anymore.

### PR DESCRIPTION
**Pull request by slack**
Initial commit for new diagrams.

Fixes #8

### CODE REVIEW

File: src/diagrams/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Architecture Diagrams
+
+This is an OmniGraffle file which holds the source materials for the following
```
Comment (bacongobbler) [inline src/diagrams/README.md]: can we make OmniGraffle a link to https://www.omnigroup.com/omnigraffle?

**Comment by slack** [comment]
Oops, had an un-pushed commit locally: https://github.com/slack/docs-v2/commit/71cb040428ca57a2e0062c90dcb4f005e591e1b1

Please review again.

Also added a link to OmniGraffle.

File: .gitignore [hunk 2]
```diff
@@ -1,20 +1 @@
...
-develop-eggs
-.installed.cfg
-lib
-lib64
-
-# python virtual environments for testing
-venv/
-
-_build/
+.DS_Store
```
Comment (bacongobbler) [inline .gitignore]: this should be part of a global gitignore: https://github.com/deis/deis/pull/2566

Comment (slack) [inline .gitignore]: wat. ok

**Comment by slack** [comment]
Rebased out the `.gitignore` commit.

**Pull request merged**